staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -900,11 +900,19 @@ func (c *Cacher) processEvent(event *watchCacheEvent) {
 	c.incoming <- *event
 }
 
-func (c *Cacher) dispatchEvents() {
+func (c *Cacher) dispatchEvents(progressRequester *conditionalProgressRequester) {
 	// Jitter to help level out any aggregate load.
 	bookmarkTimer := c.clock.NewTimer(wait.Jitter(time.Second, 0.25))
 	defer bookmarkTimer.Stop()
 
+	// TODO(p0lyn0mial): adapt the following logic once
+	//  https://github.com/kubernetes/kubernetes/pull/124612 merges
+	progressRequesterCleanUpOnceFn := func() { /*no-op*/ }
+	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) && utilfeature.DefaultFeatureGate.Enabled(features.WatchList) {
+		progressRequester.Add()
```

Comment:
I have just also realised than we need to slightly change the `progressRequester` so that it is able to send periodic progress updates. This will unblock watchers initialised from the global RV against resources that haven't received any changes/updates.

I can try changing the `progressRequester` for that purpose.

Reply:
The question is what should we do when the etcd was started with the progress notification flag ?

Reply:
actually, no, this will be handled by https://github.com/kubernetes/kubernetes/blob/b0abfdecd4b9a5a1df7d221d0d601b4aa2558871/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L1317